ext/mysqli/tests/mysqli_auth_pam.phpt

```diff
@@ -19,8 +19,8 @@ if (!$res = $link->query("SHOW PLUGINS"))
 
 $have_pam = false;
 while ($row = $res->fetch_assoc()) {
-    if (isset($row['Name']) && ('mysql_clear_password' == $row['Name'])) {
-        $have_pam = true;
+    if (isset($row['Name']) && in_array($row['Name'], array('pam', 'mysql_clear_password', 'auth_pam_compat'))) {
```

Comment:
This line of code appears to be looping through the list of server-side plugins returned by `SHOW PLUGINS`.

`mysql_clear_password` is a client-side plugin used for the following server-side plugins:

* `pam` (MariaDB Server and Percona Server)
* `auth_pam_compat` (Percona Server)
* `authentication_pam` (MySQL Enterprise Server).

As far as I know, Client-side plugins don't show up in the output of `SHOW PLUGINS`, so I don't think `mysql_clear_password` will ever show up in this loop. Should `mysql_clear_password` be removed from this specific check?

Reply:
opps. corrected to `authentication_pam`. Looked at https://dev.mysql.com/doc/refman/8.0/en/pam-pluggable-authentication.html#pam-pluggable-authentication-usage

I have pinged some MySQL community folks to test this their Enterprise side.